data_write/data_write.py

```diff
@@ -75,165 +92,244 @@ def __init__(self, processed_data, data_write_options):
-        1) pre_bf_iq
-        2) iq
-        3) rawacf
-        The file format order of preference is:
-        1) hdf5
-        2) json
-        3) dmap
+        Parse through samples and write to file. 
+
+        A file will be created using the file extention for each requested data product.
+
+        :param write_iq:        Should IQ be written to file? Bool.
+        :param write_pre_bf_iq: Should pre-beamformed IQ be written to file? Bool.
+        :param write_raw_rf:    Should raw rf samples be written to file? Bool.
+        :param file_ext:        Type of file extention to use. String
         :param write_rawacf: Should rawacfs be written to file? Bool, default True.
-        :param write_iq: Should IQ be written to file? Bool, default False.
-        :param write_pre_bf_iq: Should pre-beamformed IQ be written to file? Bool. Default False
-        :param use_hdf5: Write data out to hdf5 file. Default False
-        :param use_json: Write data out to json file. Default True
-        :param use_dmap: Write data out to dmap file. Default False
         """
-        # Find out what file format to write out
-        file_format_string = None
-        if use_hdf5:
-            file_format_string = 'hdf5'
-        elif use_json:
-            file_format_string = 'json'
-        elif use_dmap:
-            file_format_string = 'dmap'
-
-        if not file_format_string:
+
+        if file_ext not in ['hdf5','json','dmap']:
             raise ValueError("File format selection required (hdf5, json, dmap), none given")
 
-        main_iq_available = False
-        intf_iq_available = False
-        rawacf_available = False
-        pre_bf_iq_available = False
-        data_format_string = None
+        # Format the name and location for the dataset
+        today_string = datetime.datetime.today().strftime("%Y%m%d")
+        datetime_string = datetime.datetime.today().strftime("%Y%m%d.%H%M.%S.%f")
+        dataset_directory = "{0}/{1}".format(self.options.data_directory, today_string)
+        dataset_name = "{dt}.{site}.{{dformat}}.{fformat}".format(dt=datetime_string,
+                                                                site=self.options.site_id,
+                                                                fformat=file_ext)
+        dataset_location = "{dir}/{{name}}".format(dir=dataset_directory)
+
+
+        def write_file(location, final_data_dict):     
+            """
+            Writes the final data out to the location based on the type of file extention required
+
+            :param location:        File path and name to write to. String
+            :param final_data_dict: Data dict parsed out from protobuf. Dict
+                
+            """
+            if not os.path.exists(dataset_directory):
+                try:
+                    os.makedirs(dataset_directory)
+                except:
+                    pass
```

Comment:
It fails I think if the dir exists already, which is acceptable. However it might fail if there is a permissions issue or something so maybe we can take a look if there are some special exceptions to take a look at.